flink-clients/src/main/java/org/apache/flink/client/program/rest/RestClusterClient.java

```diff
@@ -104,11 +122,26 @@ protected JobSubmissionResult submitJob(JobGraph jobGraph, ClassLoader classLoad
 		} catch (JobSubmissionException e) {
 			throw new ProgramInvocationException(e);
 		}
-		// don't return just a JobSubmissionResult here, the signature is lying
-		// The CliFrontend expects this to be a JobExecutionResult
 
-		// TOOD: do not exit this method until job is finished
-		return new JobExecutionResult(jobGraph.getJobID(), 1, Collections.emptyMap());
+		final JobExecutionResult jobExecutionResult = waitForJobExecutionResult(jobGraph.getJobID());
+
+		if (jobExecutionResult.getSerializedThrowable().isPresent()) {
+			final SerializedThrowable serializedThrowable = jobExecutionResult.getSerializedThrowable().get();
+			final Throwable throwable = serializedThrowable.deserializeError(classLoader);
+			throw new ProgramInvocationException(throwable);
+		}
+
+		try {
+			// don't return just a JobSubmissionResult here, the signature is lying
+			// The CliFrontend expects this to be a JobExecutionResult
+			this.lastJobExecutionResult = new SerializedJobExecutionResult(
+				jobExecutionResult.getJobId(),
+				jobExecutionResult.getNetRuntime(),
+				jobExecutionResult.getAccumulatorResults()).toJobExecutionResult(classLoader);
```

Comment:
We could also directly use `AccumulatorHelper#deserializeAccumulators`.

Reply:
I need the `JobExecutionResult` here; not only accumulators.